weatherAplication/ShowWeather/ShowWeatherViewController.swift

```diff
@@ -15,65 +13,31 @@ class ShowWeatherViewController: UIViewController {
-                alert.addAction(UIAlertAction(title: "OK", style: .default))
-                self?.present(alert, animated: true)
-            }
-
-            if let httpResponse = response as? HTTPURLResponse {
-                if 200...299 ~= httpResponse.statusCode {
-                    do {
-                        guard let data = data else { return }
-                        let jsonResult = try JSONDecoder().decode(OpenMeteoDecoded.self, from: data)
-                        self?.takeDataFromJson(jsonResult: jsonResult)
-                        DispatchQueue.main.async {
-                            self?.weatherTableView.reloadData()
-                        }
-                    } catch let error {
-                        print(error)
-                    }
-                }
-            }
-
-        }).resume()
-    }
-
-    func takeDataFromJson(jsonResult: OpenMeteoDecoded) {
-        var dataArray: [WeatherData] = []
-        if jsonResult.hourly.time.count >= kElementsToShow,
-           jsonResult.hourly.temperature.count >= kElementsToShow,
-           jsonResult.hourly.surfacePressure.count >= kElementsToShow,
-           jsonResult.hourly.windspeed.count >= kElementsToShow,
-           jsonResult.hourly.weathercode.count >= kElementsToShow {
-            date = String(jsonResult.hourly.time[0].prefix(kCharsToDrop - 1))
-            for index in 0...(kElementsToShow - 1) {
-                dataArray.append(WeatherData(hour: String(jsonResult.hourly.time[index].dropFirst(kCharsToDrop)),
-                                             temperature: jsonResult.hourly.temperature[index],
-                                             pressure: jsonResult.hourly.surfacePressure[index],
-                                             windSpeed: jsonResult.hourly.windspeed[index],
-                                             weatherCode: jsonResult.hourly.weathercode[index]))
-            }
-            self.dataArray = dataArray
-        }
-    }
+    private let apiClient = APIClient.sharedInstance
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupView()
         setupConstraints()
-        makeURLRequest()
+        apiClient.searchWeather(latitude: latitude, longitude: longitude, completion: { [weak self] weatherData, date -> Void in
+            self?.dataArray = []
+            self?.dateString = ""
+            self?.dateString = date
+            switch weatherData {
+            case .success(let weatherData):
+                self?.dataArray = weatherData
+                DispatchQueue.main.async {
+                    self?.weatherTableView.reloadData()
+                }
+            case .failure(let error):
+                let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
+                alert.addAction(UIAlertAction(title: "OK", style: .default))
+                self?.present(alert, animated: true)
```

Comment:
zauważ że weatherData sugeruje nam że ta zmienna ma jakiś konkretny format danych- w tym przypadku to pewnie były by jakieś dane pogodowe, a zupełnie nie sugeruje że to może być enum z success i failure. Poza tym nie jest to też super czytelny. Myślę że lepszym rozwiązaniem jest oznaczenie funkcji searchWeather jako mogącej potencjalnie zfailować (poprzez nadanie jej atrybutu _throws_) i dopiero w tym miejscu zrobienie
do {
try ... 
} catch {
...
}